pkg/reboot/calculator.go

```diff
@@ -106,14 +115,79 @@ func (s *safeTimeCalculator) calcMinTimeAssumeRebooted() error {
-		s.log.Error(err, err.Error(), "requested SafeTimeToAssumeNodeRebootedSeconds", s.timeToAssumeNodeRebooted, "minimal calculated value for SafeTimeToAssumeNodeRebootedSeconds", minTime)
+	//update related logic of min time on configuration if necessary
+	if err := s.manageSafeRebootTimeInConfiguration(minTime); err != nil {
 		return err
 	}
+
 	return nil
 }
 
+// manageSafeRebootTimeInConfiguration does two things:
+//  1. It sets Status.MinSafeTimeToAssumeNodeRebootedSeconds in case it's changed by latest calculation.
+//  2. It Adds/removes config.Status.SpecSafeTimeOverriddenWarning if necessary.
+func (s *safeTimeCalculator) manageSafeRebootTimeInConfiguration(minTime time.Duration) error {
+	minTimeSec := int(minTime.Seconds())
+	config, err := s.getConfig()
+	if err != nil {
+		return err
+	}
+	orgConfig := config.DeepCopy()
+	prevMinRebootTimeSec := config.Status.MinSafeTimeToAssumeNodeRebootedSeconds
+	if prevMinRebootTimeSec != minTimeSec {
+		config.Status.MinSafeTimeToAssumeNodeRebootedSeconds = minTimeSec
+	}
+	//Manage warning
+	config.Status.SpecSafeTimeOverriddenWarning = ""
+	if config.Spec.SafeTimeToAssumeNodeRebootedSeconds > 0 && minTimeSec > config.Spec.SafeTimeToAssumeNodeRebootedSeconds {
+		config.Status.SpecSafeTimeOverriddenWarning = v1alpha1.SafeTimeToAssumeNodeRebootedSecondsWarning
+		s.log.Info("SafeTimeToAssumeNodeRebootedSeconds is overridden by calculated value because it's invalid, calculated value stored in Status.MinSafeTimeToAssumeNodeRebootedSeconds would be used instead")
+		events.WarningEvent(s.recorder, config, "SafeTimeToAssumeNodeRebootedSecondsInvalid", "SafeTimeToAssumeNodeRebootedSeconds is overridden by calculated value")
+	}
+
+	if !reflect.DeepEqual(config, orgConfig) {
+		if err := s.k8sClient.Status().Patch(context.Background(), config, client.MergeFrom(orgConfig)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *safeTimeCalculator) getMinSafeTimeSecFromConfig() (time.Duration, error) {
+	config, err := s.getConfig()
+	if err != nil {
+		return 0, err
+	}
+
+	minTimeSec := config.Status.MinSafeTimeToAssumeNodeRebootedSeconds
+	if minTimeSec > 0 {
+		return time.Duration(minTimeSec) * time.Second, nil
+	}
+
+	err = errors.New("failed getting MinSafeTimeToAssumeNodeRebootedSeconds, value isn't initialized")
+	s.log.Error(err, "MinSafeTimeToAssumeNodeRebootedSeconds should not be empty")
+	return 0, err
+}
+
+func (s *safeTimeCalculator) getConfig() (*v1alpha1.SelfNodeRemediationConfig, error) {
+	confList := &v1alpha1.SelfNodeRemediationConfigList{}
+	if err := s.k8sClient.List(context.Background(), confList); err != nil {
+		s.log.Error(err, "failed to get snr configuration")
+		return nil, err
+	}
+
+	var config v1alpha1.SelfNodeRemediationConfig
+	if len(confList.Items) < 1 {
+		err := errors.New("SNR config not found")
+		s.log.Error(err, "failed to get snr configuration")
+		return nil, err
+	} else {
+		config = confList.Items[0]
```

Comment:
not related to this PR: do we have something in place to prevent multiple configs? :thinking:

Reply:
The same thought crossed my mind briefly :sweat_smile: 

ATM the initial config is created automatically. Maybe we can do something simple such as a webhook which rejects all config creations expect the default one ?

Reply:
https://github.com/medik8s/self-node-remediation/pull/203